pandas/core/frame.py

```diff
@@ -2663,67 +2663,72 @@ def _ixs(self, i, axis=0):
     def __getitem__(self, key):
         key = com._apply_if_callable(key, self)
 
-        # shortcut if we are an actual column
-        is_mi_columns = isinstance(self.columns, MultiIndex)
+        # shortcut if the key is in columns
         try:
-            if key in self.columns and not is_mi_columns:
-                return self._getitem_column(key)
-        except:
+            if self.columns.is_unique and key in self.columns:
+                if self.columns.nlevels > 1:
+                    return self._getitem_multilevel(key)
+                return self._get_item_cache(key)
+        except (ValueError, TypeError):
             pass
```

Comment:
what hits this exception here?

Reply:
Lots of cases, for instance ``pd.DataFrame(index=range(3), columns=range(3))[['a', 'b']]``

Reply:
Yes, I guess so